lib/apiservers/engine/backends/container.go

```diff
@@ -415,6 +419,43 @@ func (c *Container) ContainerRm(name string, config *types.ContainerRmConfig) er
 	return nil
 }
 
+// checkActivePortBindings gets port bindings for the container and
+// unmaps ports if the cVM that previously bound them isn't powered on
+func (c *Container) checkActivePortBindings(vc *viccontainer.VicContainer) error {
+	for p := range vc.HostConfig.PortBindings {
+		thePort := p.Port()
+
+		cbpLock.Lock()
+		mappedCtr, mapped := containerByPort[thePort]
+		cbpLock.Unlock()
+		if !mapped {
+			continue
+		}
+
+		log.Debugf("Container %q maps port %s", mappedCtr, thePort)
+		// check state of the previously bound container with PL
+		cc := cache.ContainerCache().GetContainer(mappedCtr)
+		running, err := c.containerProxy.IsRunning(cc)
+		if err != nil {
+			return fmt.Errorf("Failed to get container %q power state: %s",
+				mappedCtr, err)
+		}
+		if running {
+			log.Debugf("Running container %q still holds port %s", mappedCtr, thePort)
+			continue
+		}
+
+		log.Debugf("Unmapping port %s for powered off container %q",
+			thePort, mappedCtr)
+		err = c.mapPorts(portmap.Unmap, cc.HostConfig, &models.EndpointConfig{}, mappedCtr)
```

Comment:
mapPorts is a relatively confusing name in this context. MapPorts supposed to map ports not unmap them, right? It looks like a small restructuring needed.